src/main/java/io/supertokens/session/Session.java

```diff
@@ -554,7 +557,8 @@ public static JsonObject getJWTData(Main main, String sessionHandle)
         io.supertokens.pluginInterface.session.SessionInfo session = StorageLayer.getSessionStorage(main)
                 .getSession(sessionHandle);
 
-        if (session == null) {
+        // If there is no session, or session is expired
+        if (session == null || session.expiry <= new Date().getTime()) {
```

Comment:
use `System.currentTimeMillis()` instead - to be consisten with other places where we generate curr time.

Reply:
Also, if you remove `|| session.expiry <= new Date().getTime()`, i believe that one of your tests will fail right - if not, then please look into it to make sure that at least one of them do?

Reply:
Without the `|| session.expiry <= new Date().getTime()` the test that checks for refresh token expired without calling revokeSession fails

Reply:
Done